src/Service/Terminal/Service_Terminal.re

```diff
@@ -67,22 +67,30 @@ module Sub = {
         let makeDebouncedDispatch = () => {
           let scheduled = ref(false);
           let latestAction = ref(None);
+          let debounceCount = ref(0);
           action => {
-            latestAction := Some(action);
-            if (! scheduled^) {
-              scheduled := true;
-
-              let _: unit => unit =
-                Revery.Tick.timeout(
-                  () => {
-                    latestAction^ |> Option.iter(dispatch);
-                    scheduled := false;
-                    latestAction := None;
-                  },
-                  Revery.Time.zero,
-                );
-              ();
-            };
+
+            if (debounceCount^ < 1) {
+              dispatch(action);
+              incr(debounceCount);
+            } else {
+              latestAction := Some(action);
+              if (! scheduled^) {
+                scheduled := true;
+
+                let _: unit => unit =
+                  Revery.Tick.timeout(
+                    () => {
+                      debounceCount := 0;
+                      latestAction^ |> Option.iter(dispatch);
+                      scheduled := false;
+                      latestAction := None;
+                    },
+                    Revery.Time.zero,
+                  );
+                ();
+              };
+            }
```

Comment:
I also don;t think this does what you want it to do. Since there's no timer to decrement the debounceCount, you've just moved the latency to a second invocation which might occur at any time in the future.

Reply:
Ya, it's totally busted! Appreciate you catching this. I've factored it out and adding some test cases to better cover this.